src/main/java/io/github/thebusybiscuit/slimefun4/core/multiblocks/MultiBlockMachine.java

```diff
@@ -105,11 +108,16 @@ public void postRegister() {
     public void load() {
         super.load();
 
-        for (ItemStack recipeItem : displayRecipes) {
-            SlimefunItem item = SlimefunItem.getByItem(recipeItem);
-
-            if (item == null || !item.isDisabled()) {
-                recipes.add(new ItemStack[] { recipeItem });
+        Preconditions.checkArgument(displayRecipes.size() % 2 == 0, "This MultiBlockMachine's display recipes were illegally modified!");
+        for (int i = 0; i < displayRecipes.size(); i += 2) {
+            ItemStack inputStack = displayRecipes.get(i);
+            ItemStack outputStack = displayRecipes.size() >= i + 2 ? displayRecipes.get(i + 1) : null;
```

Comment:
Please dont use ternary here its against guidelines and its hard to understand here.
https://github.com/Slimefun/Slimefun4/blob/master/CONTRIBUTING.md#7-style-preferences